src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+  const modifiedStates = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const type = actions[i].type;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(modifiedStates, actions[i].extraData);
+        statesArray.push({ ...modifiedStates });
+        break;
+      case 'removeProperties':
+        for (let x = 0; x < actions[i].keysToRemove.length; x++) {
+          delete modifiedStates[actions[i].keysToRemove[x]];
+        };
+        statesArray.push({ ...modifiedStates });
+        break;
+      case 'clear':
+        for (const key in modifiedStates) {
+          delete modifiedStates[key];
+        };
+        statesArray.push({ ...modifiedStates });
+        break;
```

Comment:
```suggestion
        statesArray.push({});
        break;
```

Reply:
If I push an empty Array, modifiedStates object is not empty for the next iterations.